osu.Game/Database/EFToRealmMigrator.cs

```diff
@@ -123,8 +123,23 @@ protected override void LoadComplete()
 
         private void beginMigration()
         {
+            const string backup_folder = "backups";
+
+            string backupSuffix = $"before_final_migration_{DateTimeOffset.UtcNow.ToUnixTimeSeconds()}";
+
+            // required for initial backup.
+            var realmBlockOperations = realm.BlockAllOperations();
+
             Task.Factory.StartNew(() =>
             {
+                realm.CreateBackup(Path.Combine(backup_folder, $"client.{backupSuffix}.realm"), realmBlockOperations);
+
+                // Above call will dispose of the blocking token when done.
+                // Clean up here so we don't accidentally dispose twice.
+                realmBlockOperations = null;
```

Comment:
This is still not correct. If the backup fails, `CreateBackup()` will dispose of the token (even though it threw), then the execution of this block will be aborted, and the continuation will be executed without the token ever having been nulled.